### DISCUSSION TITLE
Custom metafunctions?

### DISCUSSION THREAD

**Discussion by TerensTare** [hsutter/cppfront#1339] [Q&A] [answered]
Hi, how can I write a custom metafunction? Do I need to "hack" them into the compiler? The docs only mention the built-in ones and don't say how to add your own.

**Comment by MaxSagebaum** [comment, marked as answer]
Yes, you have to hack them into the compiler. There is the effort by @DyXel in https://github.com/hsutter/cppfront/pull/1261. But it will currently not be mereged due to time constraints of Herb. He wants to focus on core features first.
